src/strings/ops.c

```diff
@@ -522,36 +545,93 @@ MVMString * MVM_string_concatenate(MVMThreadContext *tc, MVMString *a, MVMString
                     }
                 });
             }
-
             /* Assemble the result. */
-            result->body.num_strands = strands_a + strands_b;
-            result->body.storage.strands = allocate_strands(tc, strands_a + strands_b);
+            result->body.num_strands = strands_a + strands_b + (renormalized_section_graphs ? 1 : 0);
+            result->body.storage.strands = allocate_strands(tc, result->body.num_strands);
+            /* START 1 */
             if (effective_a->body.storage_type == MVM_STRING_STRAND) {
                 copy_strands(tc, effective_a, 0, result, 0, strands_a);
             }
             else {
-                MVMStringStrand *ss = &(result->body.storage.strands[0]);
-                ss->blob_string = effective_a;
-                ss->start       = 0;
-                ss->end         = effective_a->body.num_graphs;
-                ss->repetitions = 0;
+                int index_ss_a = 0;
+                MVMStringStrand *ss_a = &(result->body.storage.strands[index_ss_a]);
+                ss_a->blob_string = effective_a;
+                ss_a->start       = 0;
+                ss_a->end         = effective_a->body.num_graphs;
+                ss_a->repetitions = 0;
             }
+            if (renormalized_section) {
+                int index_ss_re;
+                int index_ss_a = strands_a - 1;
+                /* Tweak the end of the last strand of string a. Since if a is made up of multiple strands, we can't just refer to index 0 and instead erfer to strands_a - 1 */
+                MVMStringStrand *ss_a = &(result->body.storage.strands[index_ss_a]);
+                MVMStringStrand *ss_re = NULL;
+                ss_a->end -= consumed_a;
+                /* If the strands ends up to be zero length we need to reduce the number of strand_index and also incease lost_strands so the next operation writes over it */
+                if (ss_a->start == ss_a->end)
```

Comment:
Yes it happens by changing lost_strands. Which records the strands we lost so we can correct the indexes further on in the code.